pkg/controlplane/controller.go

```diff
@@ -172,6 +174,10 @@ func (c *Controller) Start() {
 	repairClusterIPs := servicecontroller.NewRepair(c.ServiceClusterIPInterval, c.ServiceClient, c.EventClient, &c.ServiceClusterIPRange, c.ServiceClusterIPRegistry, &c.SecondaryServiceClusterIPRange, c.SecondaryServiceClusterIPRegistry)
 	repairNodePorts := portallocatorcontroller.NewRepair(c.ServiceNodePortInterval, c.ServiceClient, c.EventClient, c.ServiceNodePortRange, c.ServiceNodePortRegistry)
 
+	c.stopCh = make(chan struct{})
+	repairClusterIPs.Start(c.stopCh)
```

Comment:
Why not call these from within RunUntil?  Is it to catch events in RunOnce, I guess?  Maybe a comment here?

Reply:
Yes - exactly.

Added a comment here. Together with a TODO to refactor it to simplify the interface and make it easier to run.
Will open an issue for this when this PR merges.